### ISSUE TITLE
try expression makes condition statements fail if role_name_prefix var is not set 

### ISSUE THREAD

**Issue by iomarcovalente** [terraform-aws-modules/terraform-aws-iam#383]
## Description

Latest code release makes condition statements [here](https://github.com/terraform-aws-modules/terraform-aws-iam/blame/e46ff1f49aef00a3c2f95a719560ec86f83c3f5b/modules/iam-role-for-service-accounts-eks/main.tf#L32) fail if `role_name_prefix` var is not set.

Code reference:
https://github.com/terraform-aws-modules/terraform-aws-iam/blame/e46ff1f49aef00a3c2f95a719560ec86f83c3f5b/modules/iam-role-for-service-accounts-eks/main.tf#L10

If your request is for a new feature, please use the `Feature request` template.

- [x] ✋ I have searched the open/closed issues and my issue is not listed.

## ⚠️ Note

Before you submit an issue, please perform the following first:

1. Remove the local `.terraform` directory (! ONLY if state is stored remotely, which hopefully you are following that best practice!): `rm -rf .terraform/`
2. Re-initialize the project root to pull down modules: `terraform init`
3. Re-attempt your terraform plan or apply and check if the issue still persists

## Versions

- Module version [Required]: 5.20

- Terraform version:
Terraform `v1.2.9`
also tested with 
TF version `v1.4.6`
- Provider version(s):
```
Terraform v1.2.9
on darwin_arm64
+ provider registry.terraform.io/gavinbunney/kubectl v1.14.0
+ provider registry.terraform.io/hashicorp/aws v4.67.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.3.2
+ provider registry.terraform.io/hashicorp/helm v2.9.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.20.0
+ provider registry.terraform.io/hashicorp/local v2.4.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/tls v3.4.0
+ provider registry.terraform.io/hashicorp/vault v3.15.2
+ provider registry.terraform.io/terraform-aws-modules/http v2.4.1
```

## Reproduction Code [Required]

`locals.tf`
```
locals {
  role_name_prefix = null
}
```
in terraform console, run:
```
try(coalesce("hello", "${local.role_name_prefix}*"), null)
```
output is always `null`

## Expected behavior

the expression to return a non `null` value

## Actual behavior

it does return `null`

## Additional context

As seen [here](https://github.com/terraform-aws-modules/terraform-aws-iam/pull/378/commits/5ca15c6b85f6802b384d6ab9687d136c11ac7a5c#r1201313793)

**Comment by bryantbiggs** [comment]
please provide a reproduction that demonstrates the issue

**Comment by ViktorTverdun** [comment]
> please provide a reproduction that demonstrates the issue

For example, this is my case - creating role for cert-manager that requires role to be self assumable.

```
module "iam_role_cert_manager" {
  source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
  version = "v5.20.0"

  role_name = "AWS_CertManagerRole"

  oidc_providers = {
    main = {
      provider_arn               = "arn:aws:iam::012345678901:oidc-provider/oidc.eks.us-east-1.amazonaws.com/id/5C54DDF35ER19312844C7333374CC09D"
      namespace_service_accounts = ["cert-manager:cert-manager"]
    }
  }
  allow_self_assume_role = true

  attach_cert_manager_policy = true
}

```

Running `terraform plan` produces 
```
│ Error: Invalid template interpolation value
│
│   on .terraform/modules/iam_role_cert_manager/modules/iam-role-for-service-accounts-eks/main.tf line 33, in data "aws_iam_policy_document" "this":
│   33:         values   = ["arn:${local.partition}:iam::${local.account_id}:role${var.role_path}${local.role_name_condition}"]
│     ├────────────────
│     │ local.role_name_condition is null
│
│ The expression result is null. Cannot include a null value in a string template.
```

I think the problem is following:
`local.role_name_condition` is `null` because `try(coalesce(var.role_name, "${var.role_name_prefix}*"), null)` produces `null`. That's because `coalesce(var.role_name, "${var.role_name_prefix}*")` fails with the same error - `Cannot include a null value in a string template.` because of `"${var.role_name_prefix}*"`.
As a result `try` returns second parameter which is `null`.

**Comment by mstiri** [comment]
I've just submitted a PR to fix the issue here: #389

**Comment by antonbabenko** [comment]
This issue has been resolved in version 5.23.1 :tada:
